File: week1/alfazya021/242-ValidAnagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) {
            return false;
        }

        int[] chars = new int[256];
        int index = 0;

        while (index < s.length()) {
            chars[s.charAt(index)]++;
            chars[t.charAt(index)]--;
            index++;
        }

        for (int i : chars) {
            if (i != 0) {
                return false;
            }
        }

        return true;
    }
}

File: week1/atakantekatan/242. Valid Anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        cnts = collections.Counter(s)
        cntt = collections.Counter(t)
        for letter in cnts:
            if cnts[letter] != cntt[letter]:
                return False
        return True